addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -260,12 +247,15 @@ options.registry.carousel = options.Class.extend({
         this._rebindEvents();
     },
     /**
+     * Retargets edition if active slide is not active.
+     *
      * @override
      */
-    onFocus: function () {
-        // Needs to be done on focus, not on start, as all other options are
-        // maybe not all initialized in start
-        this.$target.trigger('slid.bs.carousel');
+    onFocus: function ($focusTarget) {
+        const $items = $focusTarget.add($focusTarget.parents());
+        if ($items.filter('.carousel-item.active').length === 0) {
```

Comment:
Use `closest` to achieve the same result without `add`, `parents` and `filter`... but in fact I am not sure about this condition, what has the `$focusTarget` you receive here to do with anything ? Could you explain ?

Reply:
I added a parameter to the onFocus method, because as discussed previously, when activating the current slide, all of its parents (including the carousel itself), also receive a focus event. This means the the carousel would activate its currently active slide, and that activation would activate all of its parents, including the carousel, and onFocus would keep firing.

the $focusTarget parameter is the element that is being activated by _activateSnippet in the editor, this way the carousel knows whether the active slide is already activated or if it should ask for it to be activated. As discussed, if activate_snippet stops reactivating a snippet when it's already active, this is no longer necessary.

I will rename the variable to $activationTarget, which is more explicit I think, but if the multiple activations on the same snippet is fixed I can also remove it entirely (and remove the extra parameter from onFocus)

Reply:
(@sdegueldre This has been merged: https://github.com/odoo/odoo/commit/0cb78fa0479a5a0bcb0b8066441d0141f991199b but onFocus is still called if an internal snippet of the active one is enabled.)

Reply:
As discussed, in the end I went with a workaround and stored the activate snippet as a property on the SnippetMenu, and I now check whether the active slide is an ancestor of the active snippet.